ic-types/src/principal.rs

```diff
@@ -351,6 +406,67 @@ mod inner {
         const HASH_LEN_IN_BYTES: usize = <<Sha224 as Digest>::OutputSize as Unsigned>::USIZE; // 28
         const MAX_LENGTH_IN_BYTES: usize = Self::HASH_LEN_IN_BYTES + 1; // 29
 
+        pub const fn new() -> Self {
+            PrincipalInner {
+                len: 0,
+                bytes: [0; Self::MAX_LENGTH_IN_BYTES],
+            }
+        }
+
+        /// Panics if the length is over `HASH_LEN_IN_BYTES`
+        ///
+        /// ```compile_fail
```

Comment:
I still would prefer to test those methods on `Principal` itself to avoid this hidden structure.

What I'm thinking of:
1. test compile-time behaviour on Principal,
2. test same runtime behaviour on PrincipalInner?

Since `Principal` is what's exported, it's okay to test `const` behaviour on it. WDYT?

Reply:
Sounds reasonable to me.